### ISSUE TITLE
Pylance/Pyright can't resolve termcolor module

### ISSUE THREAD

**Issue by dschulman-repay** [microsoft/pylance-release#4477]
Pylance throws `Import "termcolor" could not be resolved Pylance(reportMissingImports)` in response to `from termcolor import cprint`, using the Python 3.11.4 interpreter. However, the code works fine under both Windows 10 and Ubuntu Linux 23.04, and [termcolor ](https://pypi.org/project/termcolor/)is a perfectly valid module, as far as I know (currently at v. 2.3.0).

Any idea why the lookup is failing? I note that `pip` has no trouble resolving it...

**Comment by StellaHuang95** [comment]
I couldn't repro it, did you select your interpreter to the environment you installed the `termcolor` in?

Please also add the output window logs, see https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue. Thanks!

**Comment by dschulman-repay** [comment]
I found the problem.

It's weird, but Pylance resolved the above import once I trusted the workspace containing my Python script; it could not do this in Restricted mode.

I don't understand this behavior.

**Comment by rchiodo** [comment]
This is because the workspace has to be trusted in order for us to get the sys.paths. I wonder if we could try and default the sys paths when the workspace isn't trusted based on say the path to the python executable.

**Comment by dschulman-repay** [comment]
OK, but what's really puzzling was that the import Pylance couldn't resolve in Restricted mode was only one of three in my script: it also imported from `argparse` and `platform`, but didn't complain about either of those.

Is this because `argparse` and `platform` are Python builtins, while `termcolor` is an external PyPi module?

**Comment by rchiodo** [comment]
> Is this because argparse and platform are Python builtins, while termcolor is an external PyPi module?

Yep that's exactly why. The builtin libraries for Python are always found.
